File: LC/Medium/150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        
        for t in tokens:
            if t not in "+-*/":
                stack.append(int(t))
            else:
                n1 = stack.pop()
                n2 = stack.pop()
                
                if t == "+":
                    curr = n2+n1
                elif t == "-":
                    curr = n2-n1
                elif t == "*":
                    curr = n2*n1
                else:
                    curr = int(float(n2)/n1)
                stack.append(curr)
                
        return stack[0]

File: LC/C++/Medium/150.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::vector<int> stack{};

        for (const std::string& c : tokens) {
            if (c == "+" || c == "-" || c == "*" || c == "/") {
                int n1 = stack.back();
                stack.pop_back();
                int n2 = stack.back();
                stack.pop_back();

                if (c == "+") {
                    stack.push_back(n2 + n1);
                } else if (c == "-") {
                    stack.push_back(n2 - n1);
                } else if (c == "*") {
                    stack.push_back(n2 * n1);
                } else if (c == "/") {
                    stack.push_back(n2 / n1);
                }
            } else {
                stack.push_back(stoi(c));
            }
        }

        return stack[0];
    }
};